### PR TITLE
Simple frontend page and api to connect to s3 bucket

### PR DESCRIPTION
**Pull request by jengoc415**
Still have problem with aws connection. Just submitting a intermediate pull request.

### CODE REVIEW

**Review by kxrt — COMMENTED** [review]
Just some style changes required to pass checks once the AWS bucket is accessible!

File: pages/api/media/index.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import messages from '../../../utils/api-messages';
+import S3 from "aws-sdk/clients/s3";
+import axios from "axios";
+
+const s3 = new S3({
+    region: "ap-southeast-1",
```
Comment (kevinchua6) [inline pages/api/media/index.ts:7]: please use 2 spaces indent instead of 4

File: pages/api/media/index.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            // for postman usage only
+            const url = await s3.getSignedUrlPromise("deleteObject", getFileParams);
+            const result = await axios.delete(url);
+            res.status(200).json({message: messages.deleteFileSuccess});
+        } else {
+            res.status(405).end(`Method ${method} not allowed`)
+        }
+    } catch(err) {
+        console.log(err);
+        if (method == 'GET') {
```
Comment (kevinchua6) [inline pages/api/media/index.ts:55]: Good job having accurate error messages! :)

File: pages/api/media/index.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        Key: req.body,
+        Expires: 120,
+    }
+
+    try {
+        if (method === 'GET') {
+            // for postman usage only
+            const url = await s3.getSignedUrlPromise("getObject", getFileParams);
+            const result = await axios.get(url);
+            res.status(200).json({message: messages.getFilesSuccess});
```
Comment (kevinchua6) [inline pages/api/media/index.ts:34]: You don't return your result? I think it should be 
`res.status(200).json({message: messages.getFilesSuccess, data: result});`

File: pages/api/media/index.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        } else if (method === 'PUT') {
+            // for "uploads/upload" use
+            const url = await s3.getSignedUrlPromise("putObject", fileParams);
+            const data = await axios.put(url, {
+                headers: {
+                  'Content-type': req.body.type,
+                  'Access-Control-Allow-Origin': '*',
+                },
+              });
+            res.status(200).json({message: messages.uploadFileSuccess});
```
Comment (kevinchua6) [inline pages/api/media/index.ts:44]: Same for this, our convention is that we return the object that we modified

File: pages/api/media/index.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        } else if (method === 'DELETE') {
+            // for postman usage only
+            const url = await s3.getSignedUrlPromise("deleteObject", getFileParams);
+            const result = await axios.delete(url);
+            res.status(200).json({message: messages.deleteFileSuccess});
```
Comment (kevinchua6) [inline pages/api/media/index.ts:49]: this also

File: pages/uploads/index.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  return (
+    <Layout title='Upload Media'>
+      <p className='container flex p-4'>These are the files in juniors club AWS bucket. Not completed.</p>
+      {/* <Image src={data} alt='data' /> */}
+    </Layout>
+  );
+};
+
+// export async function getServerSideProps(context) {
```
Comment (kevinchua6) [inline pages/uploads/index.tsx:24]: remove

File: pages/uploads/upload.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+    setFile(null);
+    setMessage('Uploaded!');
+    console.log(data.message);
+  };
+
+  return (
+    <Layout title='Upload Media'>
+      <p className='container flex p-4'>
+        Here, you can upload media files to our very own AWS bucket. Free limit is 5GB pls do not test with huge files!!
```
Comment (kevinchua6) [inline pages/uploads/upload.tsx:30]: hahaha

**Review by kevinchua6 — CHANGES_REQUESTED** [review]

**Comment by kevinchua6** [comment]
just a note, rmb to follow commit conventions, thanks!

**Review by kevinchua6 — APPROVED** [review]

**Pull request merged**